### PR TITLE
Rename group

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: pkg/users/group-actions.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return null;
+
+    const actions = [
+        <DropdownItem key="rename-group"
+                      onClick={() => { setKebabOpen(false); rename_group_dialog(group.name) }}>
+            {_("Rename group")}
+        </DropdownItem>,
+        <DropdownSeparator key="separator" />,
+        <DropdownItem key="delete-group"
+                      className={group.uid === 0 ? "" : "delete-resource-red"}
```
Comment (cockpituous) [inline pkg/users/group-actions.jsx:43]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230310-171136-e206cb18-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/local.js [hunk 2]
```diff
@@ -97,14 +122,16 @@ function AccountsPage() {
...
+                max_gid={max_gid}
             />
         );
     } else if (path.length === 1) {
         return (
             <AccountDetails accounts={accountsInfo} groups={groupsExtraInfo} shadow={shadow || []}
                             current_user={current_user_info && current_user_info.name} user={path[0]} />
         );
-    }
+    } else return null;
```
Comment (cockpituous) [inline pkg/users/local.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230310-171136-e206cb18-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-actions.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/users/group-actions.jsx:43]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230313-121051-0077b79f-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/local.js (hunk 2 above)
Comment (cockpituous) [inline pkg/users/local.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230313-121051-0077b79f-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-actions.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/users/group-actions.jsx:43]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230313-145525-cf69ea11-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/local.js (hunk 2 above)
Comment (cockpituous) [inline pkg/users/local.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230313-145525-cf69ea11-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-actions.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/users/group-actions.jsx:43]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230314-082945-8350527b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/local.js [hunk 3]
```diff
@@ -97,14 +122,16 @@ function AccountsPage() {
...
+                max_gid={max_gid}
             />
         );
     } else if (path.length === 1) {
         return (
             <AccountDetails accounts={accountsInfo} groups={groupsExtraInfo} shadow={shadow || []}
                             current_user={current_user_info?.name} user={path[0]} />
         );
-    }
+    } else return null;
```
Comment (cockpituous) [inline pkg/users/local.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230314-082945-8350527b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by KKoukiou** [comment]
@martinpitt this are just extracted commis for renaming out of the orignal groups actions PR. Nothing else was changed.

File: pkg/users/group-actions.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
```
Comment (martinpitt) [inline pkg/users/group-actions.jsx:4]: Welcome to the past! :grinning:

File: pkg/users/group-actions.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import { delete_group_dialog } from "./delete-group-dialog.js";
+import { rename_group_dialog } from "./rename-group-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+export const GroupActions = ({ group, accounts }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+
+    if (!group.isUserCreatedGroup)
+        return null;
```
Comment (martinpitt) [inline pkg/users/group-actions.jsx:34]: FTR: I explicitly welcome dropping the ability to delete system groups. These should remain under the OS'es control.

File: pkg/users/rename-group-dialog.jsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+function RenameGroupDialogBody({ state, change }) {
+    const { name } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup fieldId="group-name" label={_("New name")}>
+                <TextInput id="group-name" onChange={val => change("name", val)} value={name} />
+            </FormGroup>
+            <HelperText>
+                <HelperTextItem variant="warning">{_("There might be permission-based side effects associated with group renaming")}</HelperTextItem>
```
Comment (martinpitt) [inline pkg/users/rename-group-dialog.jsx:39]: I'm not super happy about the text. This is primarily about sudo and polkit, right? Maybe

> Renaming a group may affect sudo and similar rules

Also, I keep forgetting -- period or not? @garrett -- any opinion about making this shorter, more useful, and easier to read and translate?

File: pkg/users/group-actions.jsx (hunk 1 above)
Comment (martinpitt) [inline pkg/users/group-actions.jsx:43]: What is that special case about? What is the "uid" of a group? And shouldn't deleting the `root` group already be prevented by the "is system group" check? That might also take care of the above coverage miss.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! One weird special case and some text bikeshedding, but by and large looks good to me.

File: pkg/users/group-actions.jsx (hunk 1 above)
Comment (KKoukiou) [reply pkg/users/group-actions.jsx:43]: This is just a copied code, but I think the condition is some wrong copy anyhow. Good catch.

File: pkg/users/local.js (hunk 3 above)
Comment (cockpituous) [inline pkg/users/local.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18486-20230314-162600-009bec2c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**